test/unit/gateways/cyber_source_test.rb

```diff
@@ -472,9 +467,69 @@ def test_successful_credit_cart_purchase_single_request_without_ignore_ccv
       true
     end.returns(successful_purchase_response)
 
-    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(
-                                                                 ignore_cvv: 'false'
-                                                               ))
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(ignore_cvv: 'false'))
+    assert_success response
+  end
+
+  def test_successful_apple_pay_purchase_subsequent_auth_visa
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      assert_not_match %r'<cavv>', request_body
+      assert_not_match %r'<xid>', request_body
+      assert_match %r'<commerceIndicator>internet</commerceIndicator>', request_body
+      true
+    end.returns(successful_purchase_response)
+
+    options = @options.merge({
+      stored_credential: {
+        initiator: 'merchant',
+        reason_type: 'unscheduled',
+        network_transaction_id: '016150703802094'
+      }
+    })
+    assert response = @gateway.purchase(@amount, @apple_pay, options)
+    assert_success response
+  end
+
+  def test_successful_apple_pay_purchase_subsequent_auth_mastercard
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      assert_not_match %r'<authenticationData>', request_body
+      assert_match %r'<commerceIndicator>internet</commerceIndicator>', request_body
+      true
+    end.returns(successful_purchase_response)
+
+    credit_card = network_tokenization_credit_card('5555555555554444',
+                                                   brand: 'master',
+                                                   transaction_id: '123',
+                                                   eci: '05',
+                                                   payment_cryptogram: '111111111100cryptogram',
+                                                   source: :apple_pay)
+    options = @options.merge({
+      stored_credential: {
+        initiator: 'merchant',
+        reason_type: 'unscheduled',
+        network_transaction_id: '016150703802094'
+      }
+    })
+    assert response = @gateway.purchase(@amount, credit_card, options)
+    assert_success response
+  end
+
+  def test_successful_network_token_purchase_subsequent_auth_visa
+    @gateway.expects(:ssl_post).with do |_host, request_body|
+      assert_match %r'<cavv>111111111100cryptogram</cavv>', request_body
+      assert_match %r'<commerceIndicator>vbv</commerceIndicator>', request_body
+      assert_not_match %r'<commerceIndicator>internet</commerceIndicator>', request_body
```

Comment:
Test to verify we do not omit certain data for recurring NT like we do for recurring AP